File: My_RxJava/app/src/main/java/com/laioffer/my_rxjava/L3_source/demo4_map变换/Demo4Activity.java
package com.laioffer.my_rxjava.L3_source.demo4_map变换;

import android.graphics.Bitmap;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.ObservableEmitter;
import io.reactivex.rxjava3.core.ObservableOnSubscribe;
import io.reactivex.rxjava3.core.Observer;
import io.reactivex.rxjava3.disposables.Disposable;
import io.reactivex.rxjava3.functions.Function;

/**
 * @description: map变换，源码分析
 * @date: 2022/9/24 8:28 下午
 */
public class Demo4Activity extends AppCompatActivity {

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    executeTask();
  }

  private void executeTask() {

    // Observable.create() 返回ObservableCreate对象
    Observable.create(new ObservableOnSubscribe<String>() {
          @Override
          public void subscribe(@NonNull ObservableEmitter<String> emitter) throws Throwable {

          }
        })

        .map(new Function<String, Bitmap>() {
          @Override
          public Bitmap apply(String s) {
            return null;
          }
        })

        .subscribe(new Observer<Bitmap>() {
          @Override
          public void onSubscribe(@NonNull Disposable d) {

          }

          @Override
          public void onNext(@NonNull Bitmap bitmap) {

          }

          @Override
          public void onError(@NonNull Throwable e) {

          }

          @Override
          public void onComplete() {

          }
        });
  }


}

File: My_Jetpack/app/src/main/java/com/laioffer/my_jetpack/lesson1_lifecycle/demo4_接口监听/Demo4Activity.kt
package com.laioffer.my_jetpack.lesson1_lifecycle.demo4_接口监听

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * @description 接口监听法
 * @date        2022/11/2 10:41 下午
 *
 * Demo4Activity是一个被观察者，且持有LifecycleOwner，专业名词:宿主，含义就是activity/fragment生命周期
 */
class Demo4Activity : AppCompatActivity() {

  private lateinit var myPresenter: IPresenter

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    // 一般是有多个IPresenter实现类的
    myPresenter = MyPresenter()

    // 观察者与被观察者关联起来
    // 被观察者[Observable].addObserver(观察者[Observer])
    lifecycle.addObserver(myPresenter)
  }
}